File: data structures and algorithms/python/08_005.py
def calculate_resilience(production):

    n = len(production)
    resilience = [1] * n
    
    # Calculate products of all elements to the left
    left_product = 1
    for i in range(n):
        resilience[i] = left_product
        left_product *= production[i]
    
    # Calculate products of all elements to the right
    # and multiply with existing values
    right_product = 1
    for i in range(n-1, -1, -1):
        resilience[i] *= right_product
        right_product *= production[i]
    
    return resilience

File: data structures and algorithms/cs/08_005.cs
using System;

public class ProductionResilience
{
   public int[] CalculateResilience(int[] production)
   {
      int n = production.Length;
      int[] resilience = new int[n];

      // Calculate products of all elements to the left
      int leftProduct = 1;
      for (int i = 0; i < n; i++)
      {
         resilience[i] = leftProduct;
         leftProduct *= production[i];
      }

      // Calculate products of all elements to the right 
      // and multiply with existing values
      int rightProduct = 1;
      for (int i = n - 1; i >= 0; i--)
      {
         resilience[i] *= rightProduct;
         rightProduct *= production[i];
      }

      return resilience;
   }
}